### PR TITLE
Muscle dev

### PR DESCRIPTION
**Pull request by markwu2001**
Created bufferedTrigger() function for mi instance. When calling this function, it returns True for a trigger and False for no trigger. This is a resource intensive function and will slow down the code depending on the buffer size. **Try to run it on its own thread** 
 and in the future, hopefully it can be ported to a microcontroller which can use faster memory for the buffer. This has been tested to work with Mark's myo sensor but can use some tuning using the buffer length and input threshold with the other setup. @jered-b please integrate into main code

### CODE REVIEW

File: Muscle_Driver/muscle.py [hunk 1]
```diff
@@ -78,7 +92,22 @@ def triggered(self):
                 self.grip_T0 = time.time()
                 return False
     
-    #hey Jered, this code is meant to be run in a loop. Am I writing this correctly?
+
+    def bufferedTrigger(self):
+        #create buffers, take mean, see if next buffer is greater by a certain value
+        for i in range(len(self.currentBufferList)):
+            self.currentBufferList[i] = self.AnalogRead()
```
Comment (jered-b) [inline Muscle_Driver/muscle.py:103]: Does this need to have a sleep() to ensure that the analogRead has changed values recently?

Comment (markwu2001) [reply Muscle_Driver/muscle.py:103]: no it reads the value on the adc in real time as fast as it can. the less delay the better

Comment (markwu2001) [reply Muscle_Driver/muscle.py:103]: OHH I just understood your comment. Nah it seems to be working fine when I tested, here are some results

image

**Comment by jered-b** [comment]
Looks good, I added one final commit as you requested to implement the new buffertrigger into reduxmainprogram

**Pull request merged**